### PR TITLE
Apply new integration tests to existing framework to identify supported features

### PR DESCRIPTION
**Pull request by mikealfare**
resolves #6911

### Problem

We need to enumerate the supported features for the 1.6.0rc1 release for materialized views.

### Solution

Apply new framework integration tests to existing framework in order to identify supported features in the existing framework. Unsupported features will skip their unit tests with a message indicating why they are skipped.

### Results

Supported:
- create materialized view
- replace materialized view
- creating a materialized view in place of a table
- creating a materialized view in place of a view
- update auto_refresh on materialized view via alter
- update dist on materialized view via drop/create
- update sort on materialized view via drop/create
- execution of `on_configuration_change` config

Unsupported:
- creating a table in place of a materialized view
- creating a view in place of a materialized view

Untested:
- update backup on materialized view via drop/create
  - it was not clear where to query this in the database to confirm the change

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by mikealfare** [comment]
Note: `dev-requirements.txt` should be updated prior to merging

File: dbt/adapters/redshift/relation_configs/materialized_view.py [hunk 1]
```diff
@@ -122,8 +122,8 @@ def parse_model_node(cls, model_node: ModelNode) -> dict:
             "mv_name": model_node.identifier,
             "schema_name": model_node.schema,
             "database_name": model_node.database,
-            "backup": model_node.config.get("backup"),
-            "autorefresh": model_node.config.get("auto_refresh"),
+            "backup": model_node.config.extra.get("backup"),
+            "autorefresh": model_node.config.extra.get("auto_refresh"),
```
Comment (McKnight-42) [inline dbt/adapters/redshift/relation_configs/materialized_view.py:126]: should this be `auto_refresh`?

Comment (mikealfare) [reply dbt/adapters/redshift/relation_configs/materialized_view.py:126]: I went back and forth on it. So Redshift doesn't use any spaces anywhere, and I've been trying to adopt the concept that adatper-specific settings should appear in adapter-specific vernacular. We'll always use `schema_name` regardless of what the database calls it because we need it to be consistent. But if Redshift decided to name it `autorefresh`, I'm going with that on the object. That being said, dbt has adopted the naming convention of snake case for everything. So we will accept it in the config as `auto_refresh` and then translate it to `autorefresh` to put it on the object.

tl;dr; No, I don't think so.

Comment (McKnight-42) [reply dbt/adapters/redshift/relation_configs/materialized_view.py:126]: fantastic thanks for the clear up makes complete sense

**Review by McKnight-42 — APPROVED** [review]
seems like this will leave us in a decent state

File: .changes/unreleased/Features-20230714-160100.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: Enumerage implemented materialized view features via integration test flags
```
Comment (colin-rogers-dbt) [inline .changes/unreleased/Features-20230714-160100.yaml:2]:
```suggestion
body: Enumerate implemented materialized view features via integration test flags  and change `autorefresh` to `auto_refresh`
```
